src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java

```diff
@@ -429,12 +450,21 @@ private ArrayNode getFirmwareVersions(String deviceModel) {
     /**
      * Get SageVue systems
      *
+     * @param reAuthenticated true if this is a second attempt of getting the devices data, with the {@link #authenticate()}
+     *                        method issued
      * @return JsonNode instance containing list of SageVue systems
      */
-    private JsonNode getSystems() throws Exception {
-        authenticate();
-        String devicesResponse = doGet(BASE_URL + "systems", String.class);
-        return objectMapper.readTree(devicesResponse);
+    private JsonNode getSystems(boolean reAuthenticated) throws Exception {
+        try {
+            String devicesResponse = doGet(BASE_URL + "systems", String.class);
+            return objectMapper.readTree(devicesResponse);
+        } catch (FailedLoginException | CommandFailureException fle) {
+            if (reAuthenticated) {
+                throw new FailedLoginException("Failed to get list of systems using SessionID: " + loginId);

```

Comment:
who populates loginId?
from what I can gather it was never initialized.

Reply:
it is initialized in authenticate() method